streaming/base/format/base/writer.py

```diff
@@ -124,7 +124,7 @@ def __init__(self,
         self.shards = []
 
         # Remove local directory if requested prior to creating writer
-        local = out if isinstance(out, str) else out[0]
+        local = os.path.expanduser(out) if isinstance(out, str) else out[0]
```

Comment:
Hey @huxuan thanks for identifying this! We should apply the same thing to `out[0]`, since `out` can be a tuple of `(local, remote)`. So just this:
```suggestion
        local = os.path.expanduser(out) if isinstance(out, str) else os.path.expanduser(out[0])
```